sqlmesh/core/plan/evaluator.py

```diff
@@ -237,6 +231,9 @@ def _push(
                 snapshots,
                 allow_destructive_snapshots=plan.allow_destructive_models,
                 deployability_index=deployability_index,
+                on_start=lambda x: self.console.start_creation_progress(
```

Comment:
This is already fine, but another idea that occurred to me is instead of passing 2 callbacks, have an abstract interface like:
```
class SnapshotCreationObserver(abc.ABC):

    def on_snapshot_list_ready(self, snapshots: ...) -> None:
    
    def on_snapshot_created(self, snapshot: ...) -> None:
```

Reply:
I think this will make the interface cleaner, and potentially easier to extend in future. So I'd prefer it.